internal/store/leaf/fsck.go

```diff
@@ -71,20 +71,35 @@ type convertedSecret interface {
 }
 
 func (s *Store) fsckCheckEntry(ctx context.Context, name string) error {
+	if err := s.fsckCheckRecipients(ctx, name); err != nil {
+		out.Warningf(ctx, "Checking recipients for %s failed: %s", name, err)
+	}
+
 	// make sure we can actually decode this secret
 	// if this fails there is no way we could fix this
-	if IsFsckDecrypt(ctx) {
-		// we need to make sure Parsing is enabled in order to parse old Mime secrets
-		ctx = ctxutil.WithShowParsing(ctx, true)
-		secret, err := s.Get(ctx, name)
-		if err != nil {
-			return fmt.Errorf("failed to decode secret %s: %w", name, err)
-		}
-		if cs, ok := secret.(convertedSecret); ok && cs.FromMime() {
-			out.Warningf(ctx, "leftover Mime secret: %s\nYou should consider editing it to re-encrypt it.", name)
-		}
+	if !IsFsckDecrypt(ctx) {
+		return nil
+	}
+
+	// we need to make sure Parsing is enabled in order to parse old Mime secrets
+	ctx = ctxutil.WithShowParsing(ctx, true)
+	sec, err := s.Get(ctx, name)
+	if err != nil {
+		return fmt.Errorf("failed to decode secret %s: %w", name, err)
+	}
+	if cs, ok := sec.(convertedSecret); ok && cs.FromMime() {
+		debug.Log("leftover Mime secret: %s\nYou should consider editing it to re-encrypt it.", name)
```

Comment:
Aren't we now re-encrypting without Mime just below?

Reply:
No, but this might deserve a comment.

We're only checking if it was converted.
The type assertion doesn't change it.

The question is more of need the check at all.

Reply:
I meant the new code on line 102, is the Set actually encrypting that back as a Mime secret?

Reply:
Set is using `sec.Bytes()`. And secparse (inside Get) never returns a MIME secret, there is not MIME secret type anymore. There is a parser, but that's returning a KV secret. So this should work as intended (i.e. write as a KV secret).